src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -1745,7 +1757,85 @@ void CompositionalMultiphaseBase::applyDirichletBC( real64 const time_n,
         } );
       } );
     }
+  } );
+}
+
+void CompositionalMultiphaseBase::keepFlowVariablesConstantDuringStep( real64 const time,
+                                                                       real64 const dt,
+                                                                       DofManager const & dofManager,
+                                                                       DomainPartition & domain,
+                                                                       CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                                                                       arrayView1d< real64 > const & localRhs ) const
+{
+  GEOSX_MARK_FUNCTION;
+
+  GEOSX_UNUSED_VAR( time, dt );
+
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&]( string const &,
+                                                               MeshLevel const & mesh,
+                                                               arrayView1d< string const > const & regionNames )
+  {
+    mesh.getElemManager().forElementSubRegions( regionNames,
+                                                [&]( localIndex const,
+                                                     ElementSubRegionBase const & subRegion )
+    {
+      globalIndex const rankOffset = dofManager.rankOffset();
+      string const dofKey = dofManager.getKey( viewKeyStruct::elemDofFieldString() );
 
+      arrayView1d< integer const > const ghostRank = subRegion.ghostRank();
+      arrayView1d< globalIndex const > const dofNumber = subRegion.getReference< array1d< globalIndex > >( dofKey );
```

Comment:
Could we define `dofNumber` as a "field" and use the `getField` pattern?
Or even a std getter?

Reply:
It may be difficult to use the `getField` pattern because `dofKey` is computed dynamically by the `DofManager`. 

However, we could add a standard getter in `ObjectManagerBase`:
```
arrayView1d< globalIndex const > getDofNumber( string const & dofKey ) const 
{
  return getReference< array1d< globalIndex > >( dofKey );
}
```
and then we would use it as:
```
arrayView1d< globalIndex const > const dofNumber = subRegion.getDofNumber( dofKey );
```
Let me know what you think, I can add it in a separate PR

Reply:
That looks more engaging!